### PR TITLE
feat(protocol-designer): add development mixpanel reporting

### LINKED ISSUE #7472: PD(analytics): Add development mixpanel reporting
Right now in PD we only track analytics in production, so we have no way of knowing our events work properly until after we deploy to production. @anthonywilleee set up a dev environment in mixpanel, so lets use that in our sandbox environments.

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

Closes #7472

# Changelog
- Add development mixpanel reporting

# Risk assessment

Low

### CODE REVIEW

File: .github/workflows/pd-test-build-deploy.yaml [hunk 1]
```diff
@@ -112,6 +112,7 @@ jobs:
       - name: 'build PD'
         env:
           OT_PD_MIXPANEL_ID: ${{ secrets.OT_PD_MIXPANEL_ID }}
+          OT_PD_MIXPANEL_DEV_ID: ${{ secrets.OT_PD_MIXPANEL_DEV_ID }}
```
Comment (shlokamin) [inline .github/workflows/pd-test-build-deploy.yaml:115]: should i rename `OT_PD_MIXPANEL_ID` to `OT_PD_MIXPANEL_PROD_ID `? or is that overkill?

Comment (shlokamin) [reply .github/workflows/pd-test-build-deploy.yaml:115]: just so it matches the naming convention of `OT_PD_MIXPANEL_DEV_ID`

**Review by Kadee80 — APPROVED** [review]
🐿️

File: .github/workflows/pd-test-build-deploy.yaml (hunk 1 above)
Comment (IanLondon) [reply .github/workflows/pd-test-build-deploy.yaml:115]: I think it's fine as-is!

**Review by IanLondon — APPROVED** [review]
looks like it's working on MP!

**Pull request merged**